src/odemis/driver/mcs2.py

```diff
@@ -0,0 +1,999 @@
+
+    def SetSpeed(self, channel, value):
+        """
+        Set the speed of the SA_CTL motion
+        value: (double) indicating speed for all axes
+        """
+        logging.debug("Setting speed to %f", value)
+        # convert value to pm/s for the controller
+        speed = int(value * 1e12)
+        self.SetProperty_i64(SA_CTLDLL.SA_CTL_PKEY_MOVE_VELOCITY, channel, speed)
+
+    def GetSpeed(self, channel):
+        """
+        Returns (double) the linear speed of the SA_CTL motion
+        """
+        # value is given in pm/s
+        speed = self.GetProperty_i64(SA_CTLDLL.SA_CTL_PKEY_MOVE_VELOCITY, channel)
+        # convert to m/s
+        return float(speed) * 1e-12
+    
+    def SetAccel(self, channel, value):
+        """
+        Set the speed of the SA_CTL motion
+        value: (double) indicating speed for all axes
+        """
+        logging.debug("Setting accel to %f", value)
+        # convert value to pm/s2 for the controller
+        accel = int(value * 1e12)
+        self.SetProperty_i64(SA_CTLDLL.SA_CTL_PKEY_MOVE_ACCELERATION, channel, accel)
+
+    def GetAccel(self, channel):
+        """
+        Returns (float) the accel of the SA_CTL motion
+        """
+        # value is given in pm/s2
+        accel = self.GetProperty_i64(SA_CTLDLL.SA_CTL_PKEY_MOVE_ACCELERATION, channel)
+        # convert to m/s
+        return float(accel) * 1e-12
+
+    def IsPoseReachable(self, pos):
+        # TODO: Write a replacement version of this function since it is no longer supported
+        # by the controller
+        return True
+
+    def stop(self, axes=None):
+        """
+        Stop the SA_CTL controller and update position
+        if axes = None, stop all axes
+        """
+        if axes is None:
+            axes = self._axis_map.keys()
+            
+        for axis_name in axes:
+            self.Stop(self._axis_map.get(axis_name))
+        
+        self._updatePosition()
+
+    def _updatePosition(self):
+        """
+        update the position VA
+        """
+        p = {}
+        try:
+            for axis_name, axis_channel in self._axis_map.items():
+                p[axis_name] = self.self.GetPosition(axis_channel)
+                
+        except SA_CTLError as ex:
+            if ex.errno == SA_CTLDLL.SA_CTL_ERROR_NOT_REFERENCED:
+                logging.warning("Position unknown because SA_CTL is not referenced")
+                p = {'x': 0, 'y': 0, 'z': 0, 'rx': 0, 'ry': 0, 'rz': 0}
+            else:
+                raise
+
+        p = self._applyInversion(p)
+        logging.debug("Updated position to %s", p)
+        self.position._set_value(p, force_write=True)
+        
+    def _updateSpeed(self):
+        """
+        update the speeds
+        """
+        s = {}
+        try:
+            for axis_name, axis_channel in self._axis_map.items():
+                s[axis_name] = self.self.GetSpeed(axis_channel)
+                
+        except SA_CTLError as ex:
```

Comment:
Drop this try/except if you do nothing with the exception. (same for all the similar functions)